teuthology/provision/maas.py

```diff
@@ -0,0 +1,441 @@
+        resp = self.do_request("/boot-resources/").json()
+        if len(resp) == 0:
+            raise RuntimeError("MaaS has no images available.")
+
+        name = "%s/%s" % (
+            self.os_type,
+            ubuntu_codenames.get(self.os_version)
+            if self.os_type == "ubuntu" else self.os_version
+        )
+        for image in resp:
+            if image["name"] == name:
+                return image
+        raise RuntimeError("MaaS has no %s image." % name)
+
+    def lock_machine(self, raise_on_error=True):
+        """Lock the machine
+
+        :param raise_on_error: Whether or not to raise an exception
+                               if the request is unsuccessful (default: True)
+        """
+        resp = self.do_request(
+            f"/machines/{self.system_id}/op-lock",
+            method="POST",
+            raise_on_error=raise_on_error
+        )
+        if resp.json():
+            if not resp.json().get("locked") and raise_on_error:
+                raise RuntimeError(
+                    "Machine %s locking failed, status: %s",
+                    self.shortname, resp.json().get("locked")
+                )
+        elif raise_on_error:
+            raise RuntimeError(
+                "Failed to lock machine %s, system_id: %s",
+                self.shortname, self.system_id
+            )
+
+    def unlock_machine(self, raise_on_error=True):
+        """Unlock the machine
+
+        :param raise_on_error: Whether or not to raise an exception
+                               if the request is unsuccessful (default: True)
+        """
+        resp = self.do_request(
+            f"/machines/{self.system_id}/op-unlock",
+            method="POST",
+            raise_on_error=raise_on_error
+        )
+        if resp.json():
+            if resp.json().get("locked") and raise_on_error:
+                raise RuntimeError(
+                    "Machine %s locking failed, status: %s",
+                    self.shortname, resp.json().get("locked")
+                )
+        elif raise_on_error:
+            raise RuntimeError(
+                "Failed to lock machine %s, system_id: %s",
+                self.shortname, self.system_id
+            )
+
+    def deploy_machine(self):
+        """Deploy the machine"""
+        image_data = self.get_image_data()
+        if image_data.get("type").lower() != "synced":
+            raise RuntimeError(
+                "MaaS image %s is not synced, current status: %s",
+                image_data.get("name"), image_data.get("type")
+            )
+        log.info(
+            "Deploying machine %s with image %s",
+            self.shortname, image_data.get("name")
+        )
+    
+        files = {
+            "distro_series": (None, image_data.get("name")),
+            "user_data": (None, self._get_user_data()),
+        }
+        resp = self.do_request(
+            f"/machines/{self.system_id}/op-deploy", method="POST", files=files
+        ).json()
```

Comment:
```suggestion
        data = self.do_request(
            f"/machines/{self.system_id}/op-deploy", method="POST", files=files
        ).json()
```
This isn't a `Response`, this is an object produced by parsing the JSON body _in_ the `Response`